docs/fedora.md

```diff
@@ -0,0 +1,228 @@
+---
+title: Using mkosi-initrd with Fedora
+SPDX-License-Identifier: LGPL-2.1-or-later
+---
+
+# Building an initramfs image
+
+Two tools are required: `mkosi`, currently a git version is required,
+and the `mkosi-initrd` repository with configuration for `mkosi`.
+
+```bash
+git clone https://github.com/keszybz/mkosi
+git clone https://github.com/keszybz/mkosi-initrd
+cd mkosi-initrd
+```
+
+We generally want to use a cache, repeated builds will be much slower without this,
+and an output directory to avoid clutter:
+```bash
+mkdir -p mkosi.cache mkosi.output
+```
+
+The initrd is built for a specific kernel version.
+`kernel-core.rpm` has the modules that we want,
+but also the `vmlinuz` image, which we don't want.
+To avoid the installation of the big package, and the additional dependencies,
+and the scriptlets that try to call `kernel-install`,
+we extract the modules from the package ourselves.
+This rpm needs to be downloaded.
+We pass `KERNEL_VERSION=…` to tell the scripts what version to install.
+
+```bash
+KVER=`uname -r`
+dnf download kernel-core-$KVER.`uname -m`
+sudo PYTHONPATH=$PWD/../mkosi python -m mkosi --default fedora.mkosi -f -o initrd-$KVER.cpio.zst --environment=KERNEL_VERSION=$KVER
```

Comment:
Again on new clean rawhide I got:
mkosi: error: argument verb: invalid choice: '\nshadow-utils' (choose from 'build', 'clean', 'help', 'summary', 'genkey', 'bump', 'serve', 'build', 'shell', 'boot', 'qemu', 'ssh')

Reply:
I'd guess that you're running some older version of mkosi code, i.e. that the PYTHONPATH is not set correctly. https://github.com/systemd/mkosi/pull/847 should help.